### ISSUE TITLE
Subsequent `ember fastboot` is failing with "directory not empty"

### ISSUE THREAD

**Issue by dustinfarris** [ember-fastboot/ember-cli-fastboot#235]
## Summary

This does not happen in a fresh app, seems to be specific to my project.

Running `ember fastboot` multiple times starts failing with this error:

```
Build successful - 7367ms.

Slowest Trees                                 | Total
----------------------------------------------+---------------------
Babel                                         | 453ms
Babel                                         | 450ms

Slowest Trees (cumulative)                    | Total (avg)
----------------------------------------------+---------------------
Babel (87)                                    | 2580ms (29 ms)
SourceMapConcat: Concat: Vendor /asset... (2) | 502ms (251 ms)
SassCompiler (6)                              | 388ms (64 ms)
SourceMapConcat: Concat: Addon JS (2)         | 378ms (189 ms)

Error: ENOTEMPTY: directory not empty, rmdir '/Users/dustin/Developer/industrymaps-web/dist/node_modules/encoding/'
    at Error (native)
```

## To reproduce (in my project)

- Run `ember fastboot`
- Visit localhost:3000 in browser
- Stop fastboot server
- Run `ember fastboot` again

The directory is not always "encoding", but it's the same error each time.

Running `rm -rf dist` fixes it once, but can be reproduced again by following above steps.

**Comment by dustinfarris** [comment]
Updated: this seems to be specific to my project

**Comment by dustinfarris** [comment]
Going to close this until I can do a better job triaging.  Possible this stems from one of the addons I have installed.

**Comment by josemarluedke** [comment]
I'm also having this problem. @dustinfarris Could you share what was the problem for you?

**Comment by dustinfarris** [comment]
@josemarluedke I was told on Slack that this is a known problem with fastboot.  It doesn't seem to affect the production build, just the local server.

**Comment by steveszc** [comment]
Also having this problem, seems like a big enough productivity killer that its worth fixing. I can try to figure out steps to reproduce if no one else has been able to figure out exactly whats causing it.

**Comment by steveszc** [comment]
@dustinfarris Did your slack conversation shed any light on what might be causing this?

**Comment by dustinfarris** [comment]
I don't remember it well sorry.  I think it was some sort of race condition but not sure at all.

ping @tomdale @rwjblue

**Comment by steveszc** [comment]
Looks like this is a duplicate of issue #246

**Comment by steveszc** [comment]
This can probably be closed again in favor of #246 and #247

**Comment by tomdale** [comment]
Sorry this is taking so long to fix, we've got some improvements to the overall way the build works that should address this but it's taking longer than I anticipated. Thanks for your patience! I will close this issue in favor of #246.
